core/chains/evm/logpoller/log_poller_test.go

```diff
@@ -418,8 +419,30 @@ func TestLogPoller_Logs(t *testing.T) {
 	assert.Equal(t, event1.Bytes(), lgs[0].Topics[0])
 }
 
+// Validate that filters stored in log_filters_table match the filters stored in memory
+func validateFiltersTable(t *testing.T, lp *logPoller, orm *ORM) {
+	filters, err := orm.LoadFilters()
+	require.NoError(t, err)
+	require.Equal(t, len(filters), len(lp.filters))
+	for name, dbFilter := range filters {
+		dbFilter := dbFilter
+		memFilter, ok := lp.filters[name]
+		require.True(t, ok)
+		assert.True(t, dbFilter.CompareTo(&memFilter),
+			fmt.Sprintf("in-memory filter %s is missing some addresses or events from db filter table", name))
+		assert.True(t, memFilter.CompareTo(&dbFilter),
+			fmt.Sprintf("db filter table %s is missing some addresses or events from in-memory filter", name))
+	}
+}
+
 func TestLogPoller_RegisterFilter(t *testing.T) {
-	lp := NewLogPoller(nil, nil, nil, 15*time.Second, 1, 1, 2, 1000)
+	lggr := logger.TestLogger(t)
+	chainID := testutils.NewRandomEVMChainID()
+	db := pgtest.NewSqlxDB(t)
+	require.NoError(t, utils.JustError(db.Exec(`SET CONSTRAINTS log_poller_filters_evm_chain_id_fkey DEFERRED`)))
```

Comment:
out of curiosity - what is the intention of this db statement? why is it needed? it seems magical, maybe consider a comment

Reply:
It's telling the db to ignore any constraint violations due to the chain id not being found in the `evm_chains` table.  I noticed that's how it's done inside `SetupTH` for two similar constraints, so just copied the same pattern:
https://github.com/smartcontractkit/chainlink/pull/8427/files#diff-05e070919c02c8a135a89fbe988cbc3f4a83736f57c98fd6290b162edef08e52L44-L45

Usually, DEFERRED wouldn't imply ignored, but I think the reason it works in this case is due to the way our unit tests are set up:  all tests are encapsulated inside a single transaction, which gets rolled back at the end... but only after all the tests have either passed or failed.